### ISSUE TITLE
Lose attribute values when using AR' accepts_nested_attributes_for

### ISSUE THREAD

**Issue by nalabjp** [makandra/active_type#175]
I found that when I use AR' `accepts_nested_attributes_for` in a subclass of ActiveType::Record, the value of the attribute of the object on the nested attribures side is lost.

### Reproduced code
```
#!/usr/bin/env ruby

require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'

  gem 'active_type', ENV.fetch('AT_VERSION', '2.3.0')
  gem 'activerecord', ENV.fetch('AR_VERSION', '6.1.7.3')
  gem 'sqlite3'
  gem 'minitest'
end

ActiveRecord::Base.establish_connection(
  adapter: 'sqlite3',
  database: ':memory:'
)

ActiveRecord::Schema.define do
  create_table "teams", force: :cascade do |t|
    t.string :name
  end

  create_table "members", force: :cascade do |t|
    t.integer :team_id
    t.string :name
  end
end

ActiveRecord::Base.connection.execute('INSERT INTO teams VALUES (1, "Ocean\'s")')
ActiveRecord::Base.connection.execute('INSERT INTO members VALUES (1, 1, "Danny")')
ActiveRecord::Base.connection.execute('INSERT INTO members VALUES (2, 1, "Rusty")')

class ApplicationRecord < ActiveRecord::Base
  self.abstract_class = true
end

class Team < ApplicationRecord; end

class Member < ApplicationRecord; end

class FormTeam < ActiveType::Record[Team]
  has_many :members, class_name: 'FormMember'
  accepts_nested_attributes_for :members
end

class FormMember < ActiveType::Record[Member]
  attribute :team_name, :string
end

### TEST ###

require 'minitest/autorun'

class ActiveTypeRecordTest < Minitest::Test
  def test_nested_attributes_update_with_multiple_attributes
    form_team_params = {
      'id' => 1,
      'members_attributes' => {
        '0' => {
          'id' => '1',
          'team_name' => "Ocean's 11"
        },
        '1' => {
          'id' => '2',
          'team_name' => "Ocean's"
        }
      }
    }
    form_team = FormTeam.new(form_team_params)

    assert_equal "Ocean's 11", form_team.members[0].team_name
    assert_equal "Ocean's", form_team.members[1].team_name
  end
end
```

### Environment
active_type : 2.3.0
active_record : 6.1.7.3
ruby : 3.0.5

### Behavior

With active_type 2.2.0, this test code succeeds.
```
$ AT_VERSION=2.2.0 ruby main.rb
-- create_table("teams", {:force=>:cascade})
   -> 0.0024s
-- create_table("members", {:force=>:cascade})
   -> 0.0001s
Run options: --seed 49448

# Running:

.

Finished in 0.008141s, 122.8350 runs/s, 245.6701 assertions/s.

1 runs, 2 assertions, 0 failures, 0 errors, 0 skips
```

However, it fails in 2.3.0.
```
$ AT_VERSION=2.3.0 ruby main.rb
-- create_table("teams", {:force=>:cascade})
   -> 0.0022s
-- create_table("members", {:force=>:cascade})
   -> 0.0001s
Run options: --seed 4907

# Running:

F

Finished in 0.008546s, 117.0138 runs/s, 117.0138 assertions/s.

  1) Failure:
ActiveTypeRecordTest#test_nested_attributes_update_with_multiple_attributes [main.rb:86]:
Expected: "Ocean's 11"
  Actual: nil

1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
```
The same problem occurs with the latest active_record version 7.0.4.3.

I think the implementation added in #167 is affecting this.
When the `form_team.members[0]` in the reproduced code is executed, it reaches the following active_record code.
https://github.com/rails/rails/blob/v6.1.7.3/activerecord/lib/active_record/associations/collection_association.rb#L326-L344
`(record.attribute_names & mem_record.attribute_names)` now includes `self.class._virtual_column_names` so that `team_name` attribute is the target of the update.
This change in behavior causes "Ocean's 11" assigned to `mem_record.team_name` to be overwritten with nil.

> Any changes made to attributes on objects in the memory array are to be preserved

I don't think the current behavior is what active_type expects, since the comments on merge_target_lists say this as well.
Is there a way to resolve this issue?

**Comment by nalabjp** [comment]
I sent #176 😁

**Comment by kratob** [comment]
Thanks! Fix is released as 2.3.3.

**Comment by nalabjp** [comment]
Thank you for releasing the new version.
